src/mappings/authz/exec.ts

```diff
@@ -0,0 +1,228 @@
+globalThis.TextEncoder = util.TextEncoder;
+globalThis.TextDecoder = util.TextDecoder;
+
+const msgUpdateParamsMap: Record<string, {
+  decode(bytes: BinaryReader | Uint8Array | any): unknown
+  toJSON(obj: unknown): unknown
+}> = {
+  '/poktroll.application.MsgUpdateParam': MsgUpdateApplicationParam,
+  '/poktroll.application.MsgUpdateParams': MsgUpdateApplicationParams,
+  '/poktroll.service.MsgUpdateParam': MsgUpdateServiceParam,
+  '/poktroll.service.MsgUpdateParams': MsgUpdateServiceParams,
+  '/poktroll.supplier.MsgUpdateParam': MsgUpdateSupplierParam,
+  '/poktroll.supplier.MsgUpdateParams': MsgUpdateSupplierParams,
+  '/poktroll.gateway.MsgUpdateParam': MsgUpdateGatewayParam,
+  '/poktroll.gateway.MsgUpdateParams': MsgUpdateGatewayParams,
+  '/poktroll.proof.MsgUpdateParam': MsgUpdateProofParam,
+  '/poktroll.proof.MsgUpdateParams': MsgUpdateProofParams,
+  '/poktroll.shared.MsgUpdateParam': MsgUpdateSharedParam,
+  '/poktroll.shared.MsgUpdateParams': MsgUpdateSharedParams,
+  '/poktroll.tokenomics.MsgUpdateParam': MsgUpdateTokenomicsParam,
+  '/poktroll.tokenomics.MsgUpdateParams': MsgUpdateTokenomicsParams,
+  '/poktroll.session.MsgUpdateParams': MsgUpdateSessionParams,
+  "/cosmos.auth.v1beta1.MsgUpdateParams": MsgUpdateAuthParams,
+  "/cosmos.bank.v1beta1.MsgUpdateParams": MsgUpdateBankParams,
+  "/cosmos.distribution.v1beta1.MsgUpdateParams": MsgUpdateDistributionParams,
+  "/cosmos.mint.v1beta1.MsgUpdateParams": MsgUpdateMintParams,
+  "/cosmos.slashing.v1beta1.MsgUpdateParams": MsgUpdateSlashingParams,
+  "/cosmos.staking.v1beta1.MsgUpdateParams": MsgUpdateStakingParams,
+  "/cosmos.consensus.v1.MsgUpdateParams": MsgUpdateConsensusParams,
+  "/cosmos.gov.v1.MsgUpdateParams": MsgUpdateGovParams,
+}
+
+export async function handleAuthzExec(msg: CosmosMessage<AuthzExecMsg>): Promise<void> {
+  await attemptHandling(msg, _handleAuthzExec, unprocessedEventHandler);
+}
+
+async function _handleAuthzExec(msg: CosmosMessage<AuthzExecMsg>): Promise<void> {
+  logger.info(`[handleAuthzExec] (tx ${msg.tx.hash}): indexing message ${msg.idx + 1} / ${msg.tx.decodedTx.body.messages.length}`);
+  logger.info(`[handleAuthzExec] (msg.msg): ${stringify(msg.msg, undefined, 2)}`);
+
+  const authzExecId = messageId(msg);
+  const typeUrl = msg.msg.typeUrl;
+  const decodedAuthzMsg = msg.msg.decodedMsg;
+  const grantee = decodedAuthzMsg.grantee;
+  const msgs = decodedAuthzMsg.msgs;
+
+  await Message.create({
+    id: authzExecId,
+    typeUrl,
+    json: stringify(msg.msg.decodedMsg),
+    transactionId: msg.tx.hash,
+    blockId: msg.block.block.id,
+  }).save();
+
+  await AuthzExec.create({
+    id: authzExecId,
+    grantee,
+    messageId: authzExecId,
+    transactionId: msg.tx.hash,
+    blockId: msg.block.block.id,
+  }).save();
+
+  for (const [i, encodedMsg] of msgs.entries()) {
+    let decodedMsg: unknown;
+
+    if (encodedMsg.typeUrl in msgUpdateParamsMap) {
+      decodedMsg = await _handleUpdateParam(encodedMsg, msg.block.block.id)
```

Comment:
#PUC that this branch of the conditional not only decodes the message payload but also calls message-specific mapping handlers to index the high-level entities contained within the payload.